src/sql/workbench/services/accountManagement/browser/accountDialog.ts

```diff
@@ -231,33 +225,7 @@ export class AccountDialog extends Modal {
 		this._addAccountButton.label = localize('accountDialog.addConnection', "Add an account");
 
 		this._register(this._addAccountButton.onDidClick(async () => {
-			const vals = Iterable.consume(this._providerViewsMap.values())[0];
-
-			let pickedValue: string | undefined;
-			if (vals.length === 0) {
-				this._notificationService.error(localize('accountDialog.noCloudsRegistered', "You have no clouds enabled. Go to Settings -> Search Azure Account Configuration -> Enable at least one cloud"));
-				return;
-			}
-			if (vals.length > 1) {
-				const buttons: IQuickPickItem[] = vals.map(v => {
-					return { label: v.view.title } as IQuickPickItem;
-				});
-
-				const picked = await this._quickInputService.pick(buttons, { canPickMany: false });
-
-				pickedValue = picked?.label;
-			} else {
-				pickedValue = vals[0].view.title;
-			}
-
-			const v = vals.filter(v => v.view.title === pickedValue)?.[0];
-
-			if (!v) {
-				this._notificationService.error(localize('accountDialog.didNotPickAuthProvider', "You didn't select any authentication provider. Please try again."));
-				return;
-			}
-
-			v.addAccountAction.run();
+			this.runAddAccountAction();
```

Comment:
It doesn't really matter in this specific case - but it's still good practice to always either return or await async functions (unless you have a specific reason for not wanting to wait on them)

Reply:
done